appsScript/HandleFormSubmit.js

```diff
@@ -0,0 +1,215 @@
+const fieldNameResolutionDict = {
+  "Service License": 'schema:license',
+  "Available Languages": 'schema:availableLanguage',
+  Version: 'schema:version',
+  "Topic Discipline": "ods:topicDiscipline",
+  "Taxonomic Range": 'schema:taxonomicRange',
+  "Documentation URL": "schema:documentation",
+  "Service Type": 'schema:serviceType',
+  Name: 'schema:name',
+  Description: 'schema:description',
+  Slogan: 'schema:slogan',
+  Logo: 'schema:logo',
+  "Date Modified": 'schema:dateModified',
+  "Terms of Service": 'schema:termsOfService',
+  "Contact Email": 'schema:email',
+  "Contact Webpage": 'schema:url',
+  "Webpage": 'schema:sameAs',
+  "Identifier of Maintainer": "schema:identifier",
+  "Full Name": "schema:name",
+  "Organisation Identifier": "schema:identifier",
+  "Organisation Legal Name": "schema:legalName",
+  "Payment Model": 'schema:url',
+  "Funder Name": 'schema:name',
+  "Code Repository": 'schema:codeRepository',
+  "Change Log": 'schema:about',
+  "Runtime Platform": 'schema:runtimePlatform',
+  Status: 'schema:creativeWorkStatus',
+  "Programming Languages": 'schema:programmingLanguage',
+  "Software License": "schema:license",
+  "Content URL": "schema:contentUrl",
+  "Media License 1": "schema:license",
+  "Media License 2": "schema:license",
+  "Media License 3": "schema:license",
+};
+const licences = Licenses();
+let maintainerIndex = 0;
+let associatedMediaIndex = 0;
+
+/**
+ * Function to return the schema name for a field, when provided with the name of the field in the form
+ */
+const ReferenceField = (fieldName) => {
+  return fieldNameResolutionDict[fieldName];
+};
+
+/**
+ * Function to add a value to the taxonomic service object
+ */
+const AddToTaxonomicService = (taxonomicService, itemTitle, itemResponse) => {
+  switch (itemTitle) {
+    case 'Available Languages':
+    case 'Taxonomic Range':
+      /* For each item separated by comma in string, push to array */
+      taxonomicService[ReferenceField(itemTitle)] = itemResponse.split(',').map(item => item.replace(' ', ''));
+
+      break;
+    case 'Service Type':
+    case 'Name':
+    case 'Description':
+    case 'Slogan':
+    case 'Logo':
+    case 'Date Modified':
+    case 'Terms of Service':
+      taxonomicService['schema:Service'][ReferenceField(itemTitle)] = itemResponse;
```

Comment:
you repeat a few strings, like "schema:Service". Maybe make these into constants